pkg/store/telemetry/telemetry.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package telemetry
```

Comment:
I would still advise against breaking this into another package unless there's a reason.

Reply:
I don't mind personally any of those.

Reply:
@matej-g I got rid of the package. Could you re-review the PR, please? 🙏